crates/shared/src/price_estimation.rs

```diff
@@ -222,6 +214,52 @@ pub struct Arguments {
     pub quote_timeout: Duration,
 }
 
+#[derive(clap::Parser)]
+pub struct CoinGecko {
+    /// The API key for the CoinGecko API.
+    #[clap(long, env)]
+    pub coin_gecko_api_key: Option<String>,
+
+    /// The base URL for the CoinGecko API.
+    #[clap(
+        long,
+        env,
+        default_value = "https://api.coingecko.com/api/v3/simple/token_price"
```

Comment:
AFAIK only the pro endpoint allows batching so it seems we need to express with config args that batching should be disabled.
There are a bunch of arguments that are only needed for the buffering implementation. Maybe it makes sense to have a way to determine based on whether or not you configured all the necessary args to enable buffering.

Reply:
@MartinquaXD really good point! :ok_hand: thanks!

Reply:
I don't think we should do that. First of all, even the default is not the PRO API, but we actually never use it / intent to use. It is set like that because it is the only one working without API Key. I don't think the backend / services is responsible to assert that the CoinGecko URL is the correct one for batching, it should be 100 % responsibility of the person who set the configuration. The services should be agnostic from these things.

Reply:
So, by default, the configuration provides a URL and enabled buffering, which won't work together, correct? Not sure what exactly stops us from using that URL with disabled buffering by default.

Reply:
Really good catch @MartinquaXD , thank you very much. I corrected it and added several test to ensure we have the correct/desired behavior.